python/ray/_raylet.pyx

```diff
@@ -1530,6 +1530,9 @@ cdef void async_set_result(shared_ptr[CRayObject] obj,
         if isinstance(result, RayTaskError):
             ray.worker.last_task_error_raise_time = time.time()
             py_future.set_exception(result.as_instanceof_cause())
+        elif isinstance(result, RayError):
+            # Directly raise exception for RayActorError
+            py_future.set_exception(result)
```

Comment:
Why not also do this for _all_ `Exception`s

Reply:
Hmm this is a good point. The objectref can only be RayError or the object itself. if it's other exception, we should fail loudly here.